ui/src/components/NewGame.js

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+
+import './NewGame.css';
+
+class NewGame extends Component {
+
+    constructor() {
+        super();
+        this.state = {
+            gameName: 'pending',
+            team1: "",
+            team2: "",
+        }
+        this.onSubmit = this.onSubmit.bind(this);
+        this.routeChange = this.routeChange.bind(this);
+    }
+
+    onChange = (e) => {
+        this.setState({ [e.target.name]: e.target.value });
+    }
+
+    routeChange = () => {
+        let path = `/game/${this.state.gameName}`; 
+        this.props.history.push(path)
```

Comment:
This `history` prop is passed in from the `HomePage` component in Home.js by using the `useHistory` method from `react-router-dom`. I haven't ever used that before but this was the best way I could find to make the React Router redirect our user to the new game page without instantiating the new game on component mount and using a basic `Link`... this change means that we will only instantiate new games in the database when someone actually clicks `Start a new Game` and not every time someone visits our webpage :)